src/sentry/sentry_apps/api/endpoints/sentry_app_components.py

```diff
@@ -76,11 +78,20 @@ def get(
                     with sentry_sdk.start_span(op="sentry-app-components.prepare_components"):
                         try:
                             SentryAppComponentPreparer(component=component, install=install).run()
-                        except APIError:
-                            errors.append(str(component.uuid))
 
-                        components.append(component)
+                        except (SentryAppIntegratorError, SentryAppError) as e:
+                            error_body: dict[str, Any] = {"detail": e.message}
```

Comment:
Would it be beneficial to move building this dict into a method on the sentry app error base class itself? This would simplify the controller code here and make it easier to test this specific behavior.

Reply:
Move the transformation of an exception to a base class method and test that instead